File: Data Structures and Algorithms/4. Recursion/Ankon Sir Problems/Common Problems/_1factorial.cpp
#include <iostream>
using namespace std;

int fact(int n) {
    if(n == 0)
        return 1;

    return fact(n - 1) * n;
}

int main() {
    int n;
    cin >> n;

    cout << "Factorial of " << n << " is: " << fact(n) << '\n';
    return 0;
}

File: Data Structures and Algorithms/4. Recursion/Ankon Sir Problems/Common Problems/_1factorial.java
import java.util.*;

class _1factorial {
    public static int fact(int n) {
        if(n == 0) return 1;
        return fact(n - 1) * n;
    }

    public static void main(String[] args) {
        int n;
        Scanner sc = new Scanner(System.in);
        n = sc.nextInt();
        System.out.println("Factorial of " + n + " is " + fact(n));
    }
}
